### PR TITLE
Use a simpler, separate form for creating a new script

### PR DESCRIPTION
**Pull request by davidsbailey**
# Description

Finishes [LP-751](https://codedotorg.atlassian.net/browse/LP-751)

There were a number of problems with the  when creating a new script. This sidesteps all of them by requiring the user to create a new script first, specifying only its name, and then editing the script in a separate step.

Once that was done, scripts/_form only had one caller so I inlined it into scripts/edit.

new-script-form

## Testing story

This change really deserves an end-to-end test, such as a UI test. Unfortunately, these are not possible to write at the moment because the test machine cannot run in levelbuilder mode. The screencap above shows the new behavior, and I also manually verified that the edit form showing the new script is save-able.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/scripts_controller.rb [hunk 1]
```diff
@@ -60,9 +60,9 @@ def new
   def create
     @script = Script.new(script_params)
     if @script.save && @script.update_text(script_params, params[:script_text], i18n_params, general_params)
-      redirect_to @script, notice: I18n.t('crud.created', model: Script.model_name.human)
+      redirect_to edit_script_url(@script), notice: I18n.t('crud.created', model: Script.model_name.human)
     else
-      render 'new'
+      render json: @script.errors
```
Comment (davidsbailey) [inline dashboard/app/controllers/scripts_controller.rb:65]: makes sure any errors are shown to the user

**Comment by davidsbailey** [comment]
@islemaster and @Erin007 , I just wanted to add that I've seen csrf tokens done a lot of different ways so if you know of a better way than the one I've got here I'm happy to follow a different example.

**Review by islemaster — APPROVED** [review]
Yes, that approach is the one I'm most familiar with and I think it's fairly easy to follow.

**Comment by Erin007** [comment]
> # Description
> Finishes [LP-751](https://codedotorg.atlassian.net/browse/LP-751)
> 
> There were a number of problems with the when creating a new script. This sidesteps all of them by requiring the user to create a new script first, specifying only its name, and then editing the script in a separate step.

A number of problem with <what?> when creating a new script...

File: dashboard/app/views/scripts/edit.html.haml [hunk 2]
```diff
@@ -1,6 +1,42 @@
 %h1= @page_title = "Editing script '#{@script.name}'"
 
-= render 'form'
+= form_for(@script || :script, @script ? {} : {url: scripts_path}) do |f|
+  - if @script && @script.errors.any?
+    #error_explanation
+      %h2
+        = pluralize(@script.errors.count, 'error')
```
Comment (Erin007) [inline dashboard/app/views/scripts/edit.html.haml:7]: What?! Cool.  I didn't know there was a `pluralize`

**Review by Erin007 — APPROVED** [review]

**Comment by davidsbailey** [comment]
> A number of problem with <what?> when creating a new script...

here are the notes I have on that:
Screen Shot 2019-11-05 at 1 58 45 PM

File: dashboard/app/views/scripts/edit.html.haml (hunk 2 above)
Comment (davidsbailey) [reply dashboard/app/views/scripts/edit.html.haml:7]: > What?! Cool. I didn't know there was a `pluralize`

I didn't either! :-)

**Pull request merged**